### PR TITLE
Lint docs

### PR DESCRIPTION
**Pull request by jayvdb**
This fixes all the warnings like

```
error: redundant explicit link target
   --> butane_core/src/lib.rs:222:25
    |
222 | /// See also [`SqlVal`][crate::SqlVal].
    |               --------  ^^^^^^^^^^^^^ explicit target is redundant
    |               |
    |               because label contains path that resolves to same destination
    |
    = note: when a link's destination is not specified,
            the label is used to resolve intra-doc links
help: remove explicit link target
    |
222 | /// See also [`SqlVal`].
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,32 +1,38 @@
+CARGO := rustup run stable cargo
```
Comment (Electron100) [inline Makefile:1]: Can you say more about why this change is necessary?

Comment (jayvdb) [reply Makefile:1]: Its not strictly necessary.  My `cargo` is the distro provided one, which is pinned, and I prefer to keep it like that because I do a lot of distro packaging.  `rustup` is how I get a real unpinned stable/nightly/etc.

Comment (jayvdb) [reply Makefile:1]: If you dont want this, I can change it to
```
CARGO := cargo +stable
```
I wrote `CARGO_NIGHTLY` so that it will work with ^ syntax and the rustup syntax.

Then I can change one line, and the rest will work for me.

**Review by Electron100 — APPROVED** [review]
